types/chai-json-pattern/index.d.ts

```diff
@@ -0,0 +1,28 @@
+// Type definitions for chai-json-pattern 1.1
+// Project: https://github.com/damian-szulc/chai-json-pattern, http://chaijs.com
+// Definitions by: Christoph Herb <https://github.com/chrishrb>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.0
+
+// <reference types="chai" />
+
+declare global {
+    namespace Chai {
+        interface Assertion {
+            matchPattern(expected: any): void;
+        }
+    }
+}
+
+declare namespace ChaiJsonPattern {
+    interface Plugin {
+        [key: string]: (target: any) => void;
+    }
+
+    interface ChaiJsonPattern extends Chai.ChaiPlugin {
+        extend(plugin: Plugin): void;
+    }
+}
+
+declare const chaiJsonPattern: ChaiJsonPattern.ChaiJsonPattern;
+export default chaiJsonPattern;
```

Comment:
```suggestion
}
declare const chaiJsonPattern: Chai.ChaiPlugin;
export default chaiJsonPattern;
export function extend(plugin: ChaiJsonPattern.Plugin): void;
```

Reply:
There is no `extend` method on default object
````
const a = require("chai-json-pattern");
console.log(a.default);
console.log(a.default.extend);
console.log(a.extend);
c:\temp\test>node a.js
[Function: chaiJsonPattern]
undefined
[Function: extend]
```

Reply:
hm, i don‘t think this is want we want here. I don‘t want to change the api, only add the correct types but if we change the export here like you suggested then I have to change the import like this?

```ts
import chaiJsonPattern, { extend } from 'chai-json-pattern';
```

In the javascript source code the API for the extend method is described like [this](https://github.com/damian-szulc/chai-json-pattern/blob/master/API.md#extendextension) and i want to keep this if possible. I think the original PR was the right choice for keeping this API or am I completely wrong here?

Reply:
nvm. you are right, i will fix this